src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -0,0 +1,396 @@
+        _filteredActions.Clear();
+        auto searchText = _SearchBox().Text();
+        const bool addAll = searchText.empty();
+
+        // If there's no filter text, then just add all the commands in order to the list.
+        // - TODO GH#6647:Possibly add the MRU commands first in order, followed
+        //   by the rest of the comamnds.
+        if (addAll)
+        {
+            // Add all the commands, but make sure they're sorted alphabetically.
+            std::vector<TerminalApp::Command> sortedCommands;
+            sortedCommands.reserve(_allActions.Size());
+
+            for (auto action : _allActions)
+            {
+                sortedCommands.push_back(action);
+            }
+            std::sort(sortedCommands.begin(),
+                      sortedCommands.end(),
+                      _compareCommandNames);
+
+            for (auto action : sortedCommands)
+            {
+                _filteredActions.Append(action);
+            }
+
+            return;
+        }
+
+        // Here, there was some filter text.
+        // Show these actions in a weighted order.
+        // - Matching the first character of a word, then the first char of a
+        //   subsequent word seems better than just "the order they appear in
+        //   the list".
+        // - TODO GH#6647:"Recently used commands" ordering also seems valuable.
+        //      * This could be done by weighting the recently used commands
+        //        heigher the more recently they were used, then weighting all
+        //        the unused commands as 1
+
+        // Use a priority queue to order commands so that "better" matches
+        // appear first in the list. The ordering will be determined by the
+        // match weight produced by _getWeight.
+        std::priority_queue<WeightedCommand> heap;
+        for (auto action : _allActions)
+        {
+            const auto weight = CommandPalette::_getWeight(searchText, action.Name());
+            if (weight > 0)
+            {
+                WeightedCommand wc;
+                wc.command = action;
+                wc.weight = weight;
+                heap.push(wc);
+            }
+        }
+
+        // At this point, all the commands in heap are matches. We want to
+        // include all these commands, but to keep some semblance of
+        // determination, let's sort the commands with the same weight
+        // alphabetically by name. Otherwise, something like ["Switch to tab 0",
+        // "Switch to tab 1", "Switch to tab 2"] might show up in any order when
+        // popped from the top of the heap.
+        std::vector<TerminalApp::Command> currentWeightedCommands;
+        int currentWeight = 0;
+
+        // This is a helper lambda to sort all the commands currently in
+        // currentWeightedCommands, and add them into _filteredActions. It will
+        // then clear currentWeightedCommands
+        auto flushCurrentCommands = [&currentWeightedCommands, this]() {
```

Comment:
What if you changed the comparison in `WeightedCommand` to
```c++
if (weight < other.weight)
{
    return true;
}
else if (weight > other.weight)
{
    return false;
}
// If weight is equal, then compare the names alphabetically.
std::wstring_view leftName{ lhs.Name() };
std::wstring_view rightName{ rhs.Name() };
return leftName.compare(rightName) < 0;
```

Reply:
![bumi-mad-genius](https://user-images.githubusercontent.com/18356694/85724869-3bb34c80-b6ba-11ea-8042-799c4eb7d37f.gif)

s/Bumi/greg904/g